src/argilla/client/feedback/integrations/huggingface/card/argilla_model_template.md

```diff
@@ -0,0 +1,245 @@
+---
+# For reference on model card metadata, see the spec: https://github.com/huggingface/hub-docs/blob/main/modelcard.md?plain=1
+# Doc / guide: https://huggingface.co/docs/hub/model-cards
+{{ card_data }}
+---
+
+# Model Card for {{ model_id | default("Model ID", true) }}
+
+<!-- Provide a quick summary of what the model is/does. -->
+
+{{ model_summary | default("", true) }}
+
+## Model training
+
+Training the model using the `ArgillaTrainer`:
+
+```python
+# Load the dataset if it's uploaded on huggingface hub, or create it from scratch:
+dataset = FeedbackDataset.from_huggingface({% if dataset_name %}"{{ dataset_name }}"{% else %}"..."{% endif %})
+
+# Create the task:
+{{ trainer_task_call }}
+
+# Create the Trainer
+trainer = ArgillaTrainer(
+    dataset=dataset,
+    task=task,
+    framework="{{ framework }}",
+    {%- if lang %}{{ "\n    " }}lang="{{ lang }}",{% endif %}
+    {%- if model_id %}{{ "\n    " }}model="{{ model_id }}",{% endif %}
+    {%- if tokenizer %}{{ "\n    " }}tokenizer={{ tokenizer }},{% endif %}
+    {%- if train_size %}{{ "\n    " }}train_size={{ train_size }},{% endif %}
+    {%- if seed %}{{ "\n    " }}seed={{ seed }},{% endif %}
+    {%- if gpu_id %}{{ "\n    " }}gpu_id={{ gpu_id }},{% endif %}
+    {%- if framework_kwargs %}{{ "\n    " }}framework_kwargs={{ framework_kwargs }},{% endif %}
+)
+
+trainer.update_config(
+    # The non default hyperparameters will be filled here
```

Comment:
is there a way to get these arguments too?

Reply:
Don't spend too much time on this.

Reply:
I'm working on this right now, but its so draft to push the commits yet. They are a bit trickier depending on the model. For example, `transformers` is almost done as the inner `trainer_kwargs` is relatively simple, but in the `trl` framework it will take some work to nicely print `config=PPOConfig(batch_size=1, ppo_epochs=1)` I think. I'm working on it, if I see that it needs too much effort I'll let you know :smiley:.

Reply:
With the current implementation we cannot write the arguments passed by the user to `update_config` in the following cases:
- `spacy`: it jdoesn't work due to how the training configuration is defined with nested dicts, the (naive) function `_updated_arguments` which grabs the user arguments cannot currently deal with nested dicts. For example, the default configuration for spacy is:
```python
{
    'dev_corpus': 'corpora.dev',
    'train_corpus': 'corpora.train',
    'batcher': {
        '@batchers': 'spacy.batch_by_words.v1',
        'discard_oversize': False,
        'tolerance': 0.2,
        'size': {'@schedules': 'compounding.v1', 'start': 100, 'stop': 1000, 'compound': 1.001, 't': 0.0},
        'get_length': None
    },
    'optimizer': {'@optimizers': 'Adam.v1', 'beta1': 0.9, 'beta2': 0.999, 'L2_is_weight_decay': True, 'L2': 0.01, 'grad_clip': 1.0, 'use_averages': False, 'eps': 1e-08, 'learn_rate': 0.001},
    'seed': '${system.seed}',
    'gpu_allocator': '${system.gpu_allocator}',
    'dropout': 0.1,
    'accumulate_gradient': 1,
    'patience': 1600,
    'max_epochs': 0,
    'max_steps': 20000,
    'eval_frequency': 200,
    'score_weights': {
        'cats_score': 1.0,
        'cats_score_desc': None,
        'cats_micro_p': None,
        'cats_micro_r': None,
        'cats_micro_f': None,
        'cats_macro_p': None,
        'cats_macro_r': None,
        'cats_macro_f': None,
        'cats_macro_auc': None,
        'cats_f_per_type': None
    },
    'frozen_components': [],
    'annotating_components': [],
    'before_to_disk': None,
    'before_update': None,
    'logger': {'@loggers': 'spacy.ConsoleLogger.v1', 'progress_bar': False}
}
```
- `trl` (when using `PPO` only), the internal configuration makes it harder to print a nice representation for this case, mainly due to `PPOConfig`, so I prefer to let this for the.